### PR TITLE
Hw roles spring

### PR DESCRIPTION
**Pull request by Dinexpod**

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -29,16 +29,26 @@ public PasswordEncoder passwordEncoder() {
...
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-        http.csrf().disable();
 
-        http.authorizeRequests()
-                .antMatchers("/resources/**", "/registration", "/login")
+        http
+                .authorizeRequests()
+                .antMatchers("/developer", "/developer/listDev").hasAnyAuthority("DEVELOPER", "ADMIN", "TEAM_LEAD")
+
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java]: empty line is redundant

File: src/main/java/mate/academy/spring/controller/DeveloperController.java [hunk 2]
```diff
@@ -27,18 +28,18 @@ public DeveloperController(DeveloperService developerService) {
...
     @RequestMapping(method = RequestMethod.GET)
     public String developer() {
         log.info("controller: DEVELOPER");
         return "developer/developer";
     }
 
+//    @RequestMapping(value = "/index", method = RequestMethod.GET)
+//    public String index() {
+//        log.info("controller: DEVELOPER INDEX");
+//        return "/index";
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/controller/DeveloperController.java]: @Dinexpod commented code?

File: src/main/java/mate/academy/spring/model/Developer.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 public class Developer {
 
     @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (boroda4436) [inline src/main/java/mate/academy/spring/model/Developer.java]: @Dinexpod what is the purpose to use AUTO not IDENTITY? please describe here how they works

Comment (Dinexpod) [reply src/main/java/mate/academy/spring/model/Developer.java]: AUTO: Hibernate selects the generation strategy based on the used dialect.
IDENTITY: Hibernate relies on an auto-incremented database column to generate the primary key.

I think that In this situation it's not matter, but AUTO will choose automatically)

File: src/main/java/mate/academy/spring/controller/DeveloperController.java (hunk 2 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/controller/DeveloperController.java]: +

File: src/main/java/mate/academy/spring/config/SecurityConfig.java (hunk 1 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/spring/config/SecurityConfig.java]: +

File: src/main/java/mate/academy/spring/model/Developer.java (hunk 3 above)
Comment (boroda4436) [reply src/main/java/mate/academy/spring/model/Developer.java]: > it's not matter
all things matters

**Pull request merged**